### PR TITLE
Add topic title icon outlet

### PR DESCRIPTION
**Pull request by angusmcleod**
There are various situations in which having an outlet within the topic `h1` tag would be advantageous. The current situation prompting this is the addition of an icon to indicate the title text is translated, which is being implemented as part of https://meta.discourse.org/t/feature-request-topic-title-translation/144714/10?u=angus

Screenshot 2022-07-25 at 18 37 10

Other potential uses:

- indication that the title represents a location (locations plugin)
- indication that the title represents an object or product
- topic status indications

### CODE REVIEW

**Comment by angusmcleod** [comment]
Just a note that the discourse-assign qunit failure here seems unconnected with this PR. If that workflow could be re-run I believe it may pass.

File: app/assets/javascripts/discourse/app/templates/topic.hbs [hunk 1]
```diff
@@ -69,6 +69,8 @@
             {{#if this.model.details.can_edit}}
               <a href {{action "editTopic"}} class="edit-topic" title={{i18n "edit"}}>{{d-icon "pencil-alt"}}</a>
             {{/if}}
+
+            <PluginOutlet @name="topic-title-icon" @tagName="" @connectorTagName="span" @args={{hash model=this.model}} />
```
Comment (pmusaraj) [inline app/assets/javascripts/discourse/app/templates/topic.hbs:73]:
```suggestion
            <PluginOutlet @name="topic-title-suffix" @tagName="" @args={{hash model=this.model}} />
```

Two suggestions in the above: 
- removing the `@connectorTagName` means that the output is a little cleaner (and it's up to plugins/themes to choose exactly what the element type will be)
- these don't have to be icons, so might be better to call the outlet something else like `topic-title-suffix`

**Review by pmusaraj — CHANGES_REQUESTED** [review]

File: app/assets/javascripts/discourse/app/templates/topic.hbs (hunk 1 above)
Comment (angusmcleod) [reply app/assets/javascripts/discourse/app/templates/topic.hbs:73]: I'm happy with those changes.

**Review by pmusaraj — APPROVED** [review]
Thanks @angusmcleod

**Comment by angusmcleod** [comment]
@pmusaraj Thank you!

**Pull request merged**